metadata/metadata.go

```diff
@@ -511,28 +521,30 @@ func (resource *featureVariantResource) Schedule() string {
 func (resource *featureVariantResource) Dependencies(lookup ResourceLookup) (ResourceLookup, error) {
 	serialized := resource.serialized
 	depIds := []ResourceID{
-		{
-			Name:    serialized.Source.Name,
-			Variant: serialized.Source.Variant,
-			Type:    SOURCE_VARIANT,
-		},
-		{
-			Name: serialized.Entity,
-			Type: ENTITY,
-		},
 		{
 			Name: serialized.Owner,
 			Type: USER,
 		},
-		{
-			Name: serialized.Provider,
-			Type: PROVIDER,
-		},
 		{
 			Name: serialized.Name,
 			Type: FEATURE,
 		},
 	}
+	if serialized.Category == pb.FeatureVariantCategory(PRE_CALCULATED) {
+		depIds = append(depIds, ResourceID{
```

Comment:
Not a huge deal, dont care for how this looks. Might look better if you have a `precalculatedDependencies(serialized)` function that returns these. You can try it out and see if it looks better or worse